Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1944,6 +1938,56 @@ private extension PurchasesOrchestrator {
 
 }
 
+// MARK: - Test Store Purchases
+
+private extension PurchasesOrchestrator {
+
+    func handlePurchase(testStoreProduct: TestStoreProduct, completion: @escaping PurchaseCompletedBlock) {
+        #if TEST_STORE
+        if self.systemInfo.isTestStoreAPIKey {
+            self.purchase(testStoreProduct: testStoreProduct, completion: completion)
+        } else {
+            self.handleTestProductNotAvailableForPurchase(completion)
+        }
+        #else
+        self.handleTestProductNotAvailableForPurchase(completion)
+        #endif // TEST_STORE
+    }
+
+    #if TEST_STORE
+    private func purchase(testStoreProduct: TestStoreProduct, completion: @escaping PurchaseCompletedBlock) {
+        self.operationDispatcher.dispatchOnMainActor {
+            do {
+                try self.testStorePurchaseHandler.purchase(product: testStoreProduct) { userConfirmed in
+                    if userConfirmed {
+                        // WIP: Implement actual test purchase completion logic
+                        // For now, we'll simulate a successful purchase
+                        completion(nil, nil, nil, false)
```

Comment:
Do we need to return the CustomerInfo for this? Though I'm ok adding that later once we add the actual implementation I think

Reply:
Yeah... either that or just fetching the latest customer info... but that's probably harder, and not something we will need to do anyway, so I think a hardcoded version is ok for now!